From 9b889ed452c6e800e3efcaf1a757cc5056cf17c8 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Wed, 7 Aug 2013 16:10:02 +0800
Subject: [PATCH 006/117] atomisp2: using pci_config_lock to seralize MMIO and
 PCI CFG access

BZ: 117952

Side effect: IUNIT MMIO access will block all PCI CFG access for all PCI
devices.

[Cherry picked from MCG tree by Andy Ross]

Change-Id: I843c894999f2586cfceef8a0045a2bae51aabb37
Signed-off-by: Bin Han <bin.b.han@intel.com>
Signed-off-by: Huang Shenbo <shenbo.huang@intel.com>
Signed-off-by: Vinod Govindapillai <vinod.govindapillai@intel.com>
Reviewed-on: http://android.intel.com:8080/121460
Tested-by: Koski, Anttu <anttu.koski@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 arch/x86/pci/common.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/x86/pci/common.c b/arch/x86/pci/common.c
index 981c2db..62d7295 100644
--- a/arch/x86/pci/common.c
+++ b/arch/x86/pci/common.c
@@ -20,6 +20,8 @@
 #include <asm/pci_x86.h>
 #include <asm/setup.h>
 
+#include <linux/module.h>
+
 unsigned int pci_probe = PCI_PROBE_BIOS | PCI_PROBE_CONF1 | PCI_PROBE_CONF2 |
 				PCI_PROBE_MMCONF;
 
@@ -80,6 +82,7 @@ struct pci_ops pci_root_ops = {
  * configuration space.
  */
 DEFINE_RAW_SPINLOCK(pci_config_lock);
+EXPORT_SYMBOL(pci_config_lock);
 
 static int can_skip_ioresource_align(const struct dmi_system_id *d)
 {
-- 
1.9.3

